Mathlib/Data/Polynomial/Degree/Definitions.lean

```diff
@@ -1306,6 +1312,21 @@ theorem degree_sub_le_of_le {a b : WithBot ℕ} (hp : degree p ≤ a) (hq : degr
     degree (p - q) ≤ max a b :=
 (p.degree_sub_le q).trans <| max_le_max ‹_› ‹_›
 
+theorem leadingCoeff_sub_of_degree_lt [Ring R] (p q : R[X])
+    (h : Polynomial.degree q < Polynomial.degree p) : (p - q).leadingCoeff = p.leadingCoeff := by
+  rw [sub_eq_add_neg]
+  apply leadingCoeff_add_of_degree_lt_left
+  rw [degree_neg]
+  exact h
```

Comment:
```suggestion
theorem leadingCoeff_sub_of_degree_lt (h : Polynomial.degree q < Polynomial.degree p) :
    (p - q).leadingCoeff = p.leadingCoeff := by
  rw [sub_eq_add_neg]
  rw [← q.degree_neg] at h
  exact leadingCoeff_add_of_degree_lt' h
```
Also if we're going to have both the left and right version of `leadingCoeff_add_of_degree_lt` then we should do the same for the `sub` version.